pkg/resources/external_volume.go

```diff
@@ -0,0 +1,711 @@
+			}
+
+			// Encryption fields are set on describe requests, although they are not a valid parameter.
+			// Allowing optional checking of these fields as they will be present in some cases.
+			if withAzureEncryptionCheck {
+				encryption_type, ok := storageLocationConfig["encryption_type"].(string)
+				if ok && len(encryption_type) > 0 {
+					return nil, fmt.Errorf("unable to extract storage location, encryption_type provided for azure storage location")
+				}
+
+				encryption_kms_key_id, ok := storageLocationConfig["encryption_kms_key_id"].(string)
+				if ok && len(encryption_kms_key_id) > 0 {
+					return nil, fmt.Errorf("unable to extract storage location, encryption_kms_key_id provided for azure storage location")
+				}
+			}
+
+			azure_tenant_id, ok := storageLocationConfig["azure_tenant_id"].(string)
+			if !ok || len(azure_tenant_id) == 0 {
+				return nil, fmt.Errorf("unable to extract storage location, missing azure_tenant_id provider key in an azure storage location")
+			}
+
+			storageLocation = sdk.ExternalVolumeStorageLocation{
+				AzureStorageLocationParams: &sdk.AzureStorageLocationParams{
+					Name:           name,
+					AzureTenantId:  azure_tenant_id,
+					StorageBaseUrl: storage_base_url,
+				},
+			}
+		}
+		storageLocations[i] = storageLocation
+	}
+	return storageLocations, nil
+}
+
+func CreateContextExternalVolume(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+
+	name := d.Get("name").(string)
+	id := sdk.NewAccountObjectIdentifier(name)
+
+	storageLocations, err := extractStorageLocations(d.Get("storage_location"), true)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error creating external volume %v err = %w", id.Name(), err))
+	}
+
+	req := sdk.NewCreateExternalVolumeRequest(id, storageLocations)
+
+	if v, ok := d.GetOk("comment"); ok {
+		req.WithComment(v.(string))
+	}
+
+	if v := d.Get("allow_writes").(string); v != BooleanDefault {
+		parsed, err := booleanStringToBool(v)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		req.WithAllowWrites(parsed)
+	}
+
+	createErr := client.ExternalVolumes.Create(ctx, req)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error creating external volume %v err = %w", id.Name(), createErr))
+	}
+
+	d.SetId(helpers.EncodeResourceIdentifier(id))
+	return ReadContextExternalVolume(false)(ctx, d, meta)
+}
+
+func ReadContextExternalVolume(withExternalChangesMarking bool) schema.ReadContextFunc {
+	return func(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+		client := meta.(*provider.Context).Client
+		id := helpers.DecodeSnowflakeID(d.Id()).(sdk.AccountObjectIdentifier)
+
+		if err := d.Set("name", id.Name()); err != nil {
+			return diag.FromErr(err)
+		}
+
+		externalVolume, err := client.ExternalVolumes.ShowByID(ctx, id)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		if withExternalChangesMarking {
```

Comment:
As discussed on Slack, I’m not exactly sure how this works. It seems to be used for all Boolean values so I’ve copied it over and the tests are passing, would be good if you could check it.